jerry-core/ecma/base/ecma-gc.c

```diff
@@ -295,73 +295,110 @@ ecma_gc_mark_promise_object (ecma_extended_object_t *ext_object_p) /**< extended
 
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_PROMISE) */
 
-#if ENABLED (JERRY_ES2015_BUILTIN_CONTAINER)
-
+#if ENABLED (JERRY_ES2015_BUILTIN_MAP)
 /**
- * Mark objects referenced by Map/Set built-in.
+ * Mark objects referenced by Map built-in.
  */
 static void
-ecma_gc_mark_container_object (ecma_object_t *object_p) /**< object */
+ecma_gc_mark_map_object (ecma_object_t *object_p) /**< object */
 {
+  JERRY_ASSERT (object_p != NULL);
+
   ecma_extended_object_t *map_object_p = (ecma_extended_object_t *) object_p;
-  ecma_object_t *internal_obj_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_object_t,
-                                                                   map_object_p->u.class_prop.u.value);
-  ecma_extended_object_t *internal_ext_p = (ecma_extended_object_t *) internal_obj_p;
+  ecma_collection_t *container_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_collection_t,
+                                                                    map_object_p->u.class_prop.u.value);
+  ecma_value_t *start_p = ECMA_CONTAINER_START (container_p);
+  uint32_t entry_count = ECMA_CONTAINER_ENTRY_COUNT (container_p);
 
-  ecma_gc_set_object_visited (internal_obj_p);
+  for (uint32_t i = 0; i < entry_count; i+= ECMA_CONTAINER_PAIR_SIZE)
+  {
+    ecma_value_t *entry_p = start_p + i;
```

Comment:
How about changing this to an `ecma_container_pair_t *`?

Reply:
Yes, that would be great! :)